pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -249,12 +250,26 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	return newReconciledNormal(kc.Namespace, kc.Name)
 }
 
-func (r *Reconciler) reconcileSubscribers(ctx context.Context, ch *v1beta1.KafkaChannel) error {
+func (r *Reconciler) reconcileSubscribers(ctx context.Context, ch *v1beta1.KafkaChannel, kafkaClient sarama.Client, kafkaClusterAdmin sarama.ClusterAdmin) error {
 	after := ch.DeepCopy()
 	after.Status.Subscribers = make([]v1.SubscriberStatus, 0)
+	logger := logging.FromContext(ctx)
 	for _, s := range ch.Spec.Subscribers {
```

Comment:
A good optimization once we get this working is to do this only for the diff between `ch.Spec.Subscribers` and ready ones of `ch.Status.Subscribers` . Cause now with the smallest change in the channel, you're hitting kafka with api calls for every single subscription. 

Or in `reconcileInitialOffset` you can check if the sub is not ready then initialize the offsets.

Reply:
I think I like the suggestion for doing the check inside the `reconcileInitialOffset`

Reply:
Done now